### PR TITLE
GH-104898: Add __slots__ to os.PathLike

### PR DESCRIPTION
**Pull request by barneygale**
* Issue: gh-104898

### CODE REVIEW

File: Lib/test/test_os.py [hunk 1]
```diff
@@ -4640,6 +4640,12 @@ class A(os.PathLike):
     def test_pathlike_class_getitem(self):
         self.assertIsInstance(os.PathLike[bytes], types.GenericAlias)
 
+    def test_pathlike_subclass_slots(self):
+        class A(os.PathLike):
+            __slots__ = ()
+            def __fspath__(self):
+                return ''
+        self.assertFalse(hasattr(A(), '__dict__'))
```
Comment (brettcannon) [inline Lib/test/test_os.py:4648]: Does this need to be so fancy, or could you just do `hasattr(os.PathLike.__slots__)`?

**Review by brettcannon — APPROVED** [review]

File: Lib/test/test_os.py (hunk 1 above)
Comment (merwok) [reply Lib/test/test_os.py:4648]: I think it is testing on purpose that subclasses do inherit the slots definition.

(And you meant `hasattr(os.PathLike, '__slots__')` 😉)

Comment (barneygale) [reply Lib/test/test_os.py:4648]: If we only check for `__slots__`, we miss cases where a superclass of `os.PathLike` does not define `__slots__`:

```python
>>> class A:
...     pass
... 
>>> class B(A):
...     __slots__ = ()
... 
>>> B.__slots__
()
>>> B().__dict__
{}
```

**Comment by barneygale** [comment]
Thanks @brettcannon! Do you think this should be backported?

**Comment by merwok** [comment]
It’s not clearly a bugfix, so no.

**Pull request merged**